### PR TITLE
Registration page

### PR DESCRIPTION
**Pull request by dmohs**
How to test:
1) Register with some account.
2) Sign-in to a UI with adm.firec.

Then:
```
ACCESS_TOKEN=[token from UI]
AUTH_HEADER="Authorization: Bearer $ACCESS_TOKEN"
curl 'https://sam.dsde-dev.broadinstitute.org/api/admin/user/email/[email goes here]' -H "$AUTH_HEADER"
curl 'https://sam.dsde-dev.broadinstitute.org/api/admin/user/[subject ID from previous call]' -H "$AUTH_HEADER" -X DELETE
```

### CODE REVIEW

File: src/components/AuthContainer.js [hunk 1]
```diff
@@ -1,16 +1,52 @@
...
 
 
+function getUserIsRegisteredAndEnabled() {
+  return Sam.getUserStatus().then(response => {
+    if (response.status === 404) {
+      return false
+    } else if (!response.ok) {
+      throw response
+    } else {
+      return response.json().then(({ enabled: { ldap } }) => ldap)
```
Comment (dmohs) [inline src/components/AuthContainer.js:19]: https://app.zenhub.com/workspace/o/databiosphere/saturn-ui/issues/148

File: src/libs/ajax.js [hunk 2]
```diff
@@ -240,3 +246,16 @@ export const Dockstore = {
   }
 
 }
+
+
+export const Orchestration = {
+  profile: () => {
```
Comment (panentheos) [inline src/libs/ajax.js]: What is this closure providing?

File: src/libs/ajax.js [hunk 3]
```diff
@@ -240,3 +246,16 @@ export const Dockstore = {
...
+    return {
+      set: async keysAndValues => {
+        const url = `${await Config.getOrchestrationUrlRoot()}/register/profile`
+        await fetchOk(url, _.merge(authOpts(), jsonBody(keysAndValues), { method: 'POST' })
```
Comment (panentheos) [inline src/libs/ajax.js]: nit: Technically the async/await is unneeded here, as you could just do `return fetchOk(...)`

File: src/libs/utils.js [hunk 4]
```diff
@@ -110,3 +110,7 @@ export const memoizeWithTimeout = (fn, resolver, ms) => {
     return value
   }
 }
+
+export const delay = ms => {
```
Comment (panentheos) [inline src/libs/utils.js:114]: nit: Nice util, but is it needed?

File: src/components/AuthContainer.js [hunk 5]
```diff
@@ -54,33 +84,70 @@ export default class AuthContainer extends Component {
...
+      programLocationState: 'N/A',
+      programLocationCountry: 'N/A',
+      pi: 'N/A',
+      nonProfitStatus: 'N/A'
+    }
+    const firstName = this.state.givenName
+    const lastName = this.state.familyName
+    const contactEmail = this.state.email
+    await Orchestration.profile().set(_.merge(blankProfile, { firstName, lastName, contactEmail }))
+    this.setState({ busy: false, isRegistered: true })
```
Comment (panentheos) [inline src/components/AuthContainer.js]: What happens if registration fails?

**Review by panentheos — APPROVED** [review]
Looks good, see comments.

File: src/components/AuthContainer.js (hunk 5 above)
Comment (dmohs) [reply src/components/AuthContainer.js]: Silent error. I'm hoping that #127 is forthcoming. Should we still handle these in the meantime?

File: src/libs/utils.js (hunk 4 above)
Comment (dmohs) [reply src/libs/utils.js:114]: No, but I have written it twice so far to use while testing the UI before I added a real AJAX call. This time I decided to not delete it. Reasonable?

File: src/libs/ajax.js (hunk 3 above)
Comment (dmohs) [reply src/libs/ajax.js]: Good call.

File: src/libs/utils.js (hunk 4 above)
Comment (panentheos) [reply src/libs/utils.js:114]: Fair enough

File: src/components/AuthContainer.js (hunk 5 above)
Comment (panentheos) [reply src/components/AuthContainer.js]: That's fine for now, and the global error handling should cover it.

**Pull request merged**